### PR TITLE
Add Quad Meshing Feature to SimpleHexagonGenerator

### LINKED ISSUE #19352: Adding Meshing Capability with Quadrilateral Elements to SimpleHexagonGenerator
## Reason
`SimpleHexagonGenerator` has already been implemented as part of the `Reactor` module to enable simple meshing with triangular elements for a hexagon. There should also be simple meshing capability with quadrilateral elements for a hexagon. 

## Design
Add meshing with quadrilateral elements as an option of `SimpleHexagonGenerator`

## Impact
Allow coarser meshing for hexagons.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #19352

### CODE REVIEW

**Comment by miaoyinb** [comment]
@eshemon @shikhar413

**Comment by miaoyinb** [comment]
@loganharbour @GiudGiud This is a minor update for `SimpleHexGenerator`. Thanks in advance.

File: modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C [hunk 1]
```diff
@@ -18,6 +18,9 @@ InputParameters
 SimpleHexagonGenerator::validParams()
 {
   InputParameters params = PolygonMeshGeneratorBase::validParams();
+  MooseEnum element_type("TRI QUAD", "TRI");
+  params.addParam<MooseEnum>(
+      "element_type", element_type, "Whether the simple hexgon is made of TRI or QUAD elements.");
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C:23]:
```suggestion
      "element_type", element_type, "Whether the simple hexagon mesh is made of TRI or QUAD elements.");
```

File: modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C [hunk 2]
```diff
@@ -125,5 +151,6 @@ SimpleHexagonGenerator::generate()
         _external_boundary_name;
   }
 
+  mesh->prepare_for_use();
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C:154]: why do you need this?

**Review by GiudGiud — DISMISSED** [review]

File: modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C (hunk 2 above)
Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C:154]: We expect the mesh produced by this MG will be used by `PatternedHexMeshGenerator` as input. `PatternedHexMeshGenerator` uses `libMesh::ReplicatedMesh::stitch_meshes()` for mesh stitching. In debug mode, `libMesh::ReplicatedMesh::stitch_meshes()` checks whether `_is_prepared` data of the input mesh. I use `prepare_for_use()` here just to set `_is_prepared` as true to avert the error in debug mode. From exodus file, it seems not make a difference with or without this line. Another option is to put `prepare_for_use()` in `PatternedHexMeshGenerator` right before calling `libMesh::ReplicatedMesh::stitch_meshes()`.

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/SimpleHexagonGenerator.C:154]: ah ok. Maybe it needs a renumbering or the cell neighbors for stitching. i dont really see why right now

File: modules/reactor/test/tests/meshgenerators/simple_hexagon_generator/tests [hunk 3]
```diff
@@ -1,12 +1,22 @@
 [Tests]
-  [./simple_hexagon_generator]
+  design = 'meshgenerators/SimpleHexagonGenerator.md'
```
Comment (loganharbour) [inline modules/reactor/test/tests/meshgenerators/simple_hexagon_generator/tests:2]: Use the requirement/detail syntax here:

```
[Tests]
  design = 'meshgenerators/SimpleHexagonGenerator.md'
  issues = '#18628 #19352'

  [test]
    [tri]
      type = 'Exodiff'
      input = 'sim_hex.i'
      cli_args = '--mesh-only sim_hex_tri_in.e'
      exodiff = 'sim_hex_tri_in.e'
      recover = false

      detail = 'with TRI elements'
    []
    
    [quad]
      type = 'Exodiff'
      input = 'sim_hex.i'
      cli_args = 'Mesh/shg/element_type=QUAD
                  --mesh-only sim_hex_quad_in.e'
      exodiff = 'sim_hex_quad_in.e'
      recover = false

      detail = 'and with QUAD elements'
    []
  []
[]
```

**Review by loganharbour — CHANGES_REQUESTED** [review]
Just a small change to the test spec. Good otherwise

**Review by loganharbour — APPROVED** [review]

**Comment by miaoyinb** [comment]
Thanks @loganharbour @GiudGiud

**Pull request merged**